### PR TITLE
[cxxmodules] Introduce fCxxModulesEnabled private member

### PR DESCRIPTION
**Pull request by yamaguchi1024**
This is a better implementation over doing #ifdef R__USE__CXXMODULES
everywhere.

### CODE REVIEW

File: core/metacling/src/TCling.h [hunk 1]
```diff
@@ -111,6 +111,7 @@ class TCling : public TInterpreter {
    TObjArray*      fRootmapFiles;     // Loaded rootmap files.
    Bool_t          fLockProcessLine;  // True if ProcessLine should lock gInterpreterMutex.
    Bool_t          fAllowLibLoad;     // True if library load is allowed (i.e. not in rootcling)
+   Bool_t          fHasCxxModule;     // True if C++ modules was enabled
```
Comment (vgvassilev) [inline core/metacling/src/TCling.h]: Could you rename to `fCxxModulesEnabled = false` and just flip the flag if `R__USE_CXXMODULES` is set or `getenv("ROOT_MODULES") != nullptr`?

Comment (yamaguchi1024) [reply core/metacling/src/TCling.h]: What's the problem with fHasCxxModule ?

Comment (vgvassilev) [reply core/metacling/src/TCling.h]: It has no meaning to me. What is has cxxmodule? Does it mean it was built or ROOT is intending to use it? Also the name is inconsistent to the comment next to it ;)

**Comment by yamaguchi1024** [comment]
@phsft-bot build!

**Comment by yamaguchi1024** [comment]
What!? So it was actually this change which was upsetting fedora...

**Pull request merged**